zerver/views/custom_profile_fields.py

```diff
@@ -0,0 +1,93 @@
+from __future__ import absolute_import
+
+from typing import Text, Union, List, Dict
+import logging
+
+from django.core.exceptions import ValidationError
+from django.db import IntegrityError, connection
+from django.http import HttpRequest, HttpResponse
+from django.utils.translation import ugettext as _
+
+from zerver.decorator import has_request_variables, REQ, require_realm_admin
+from zerver.forms import ProfileFieldForm, ProfileFieldValueForm
+from zerver.lib.actions import (do_add_realm_custom_profile_field,
+                                do_remove_realm_custom_profile_field,
+                                do_update_realm_custom_profile_field,
+                                do_update_user_custom_profile_data)
+from zerver.lib.response import json_success, json_error
+from zerver.lib.validator import check_dict, check_list, check_int, check_string
+
+from zerver.models import (custom_profile_fields_for_realm, UserProfile,
+                           CustomProfileField, custom_profile_fields_for_realm)
+
+def list_realm_custom_profile_fields(request, user_profile):
+    # type: (HttpRequest, UserProfile) -> HttpResponse
+    fields = custom_profile_fields_for_realm(user_profile.realm_id)
+    return json_success({'custom_fields': [f.as_dict() for f in fields]})
+
+@require_realm_admin
+@has_request_variables
+def create_realm_custom_profile_field(request, user_profile, name=REQ(),
+                                      field_type=REQ(validator=check_int)):
+    # type: (HttpRequest, UserProfile, Text, int) -> HttpResponse
+    form = ProfileFieldForm(data={'name': name, 'field_type': field_type})
+    if not form.is_valid():
+        return json_error(_("Invalid data supplied."), data=form.errors)
```

Comment:
I'm not a fan of the Django Forms API (I find it makes the input validation code much harder to read and think about, by splitting it between two places).  So I'd probably be happiest if we just moved the validation checks into helper functions in this file like we do for most of our other views.

But if we're not going to do that, we should at least have a comment here making clear what we expect the forms part of the code to check, here, so that it's clear which part of the process is doing what and that nothing is being left behind.